### PR TITLE
[VEGA-6253] cells with warnings functionality

### PR DESCRIPTION
**Pull request by Alexandr1111**
## Задача
Cсылка в Jira GPN на задачу: https://jira.konakov.tv/browse/VEGA-6253

## Problem statement
Добавлена возможность подсветки ячеек(и их обработчики), имеющих варнинги.
https://artcpt.atlassian.net/browse/VEGA-6253
## Solution details

_{What has been done (list of details)}_

## Type
- [x] Feature
- [ ] Bug Fix
- [ ] Refactoring
- [ ] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control
- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [x] PR: Assignee chosen and necessary labels are set
- [x] PR: Current form is filled out (where it makes sense)
- [x] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [the rules](../docs/commits-style.md)
- [x] JS: There's no errors/warnings in the browser dev console
- [x] Layout: Uses [variables](https://github.com/gpn-prototypes/ui-kit/tree/master/src/components/Theme) and [CSS classes](https://github.com/gpn-prototypes/ui-kit/blob/master/src/utils/whitepaper/whitepaper.css) from [GPN design system](https://github.com/gpn-prototypes/ui-kit)
- [ ] Layout: Tested with various content amounts
- [ ] Docs: All changes to the component API are reflected and key points are described
- [ ] Storybook: Stories are written/updated for components
- [ ] Storybook: Stories follow [the rules](https://github.com/gpn-prototypes/vega-ui/blob/master/docs/storybook.md)
- [ ] Tests: New features and bugfixes are covered with tests
- [x] Tests: Existing tests passed successfully

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

File: src/components/vega-table/VegaTable.stories.tsx [hunk 1]
```diff
@@ -51,6 +60,12 @@ storiesOf('ui/VegaTable', module)
             return 32;
           }}
           removeColumn={() => {}}
+          cellsWithWarnings={cellsWithWarningsState}
+          setCellsWithWarnings={(cellsData) => {
+            // eslint-disable-next-line no-console
+            console.log('cellsWithWarnings', cellsData);
+            setCellsWithWarningsState(cellsData);
+          }}
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.stories.tsx:68]: почистить лишние пропсы

Comment (SergeyKatugin) [reply src/components/vega-table/VegaTable.stories.tsx:68]: Не почистил

File: src/components/vega-table/VegaTable.stories.tsx [hunk 2]
```diff
@@ -154,6 +172,12 @@ storiesOf('ui/VegaTable', module)
 
             return 32;
           }}
+          cellsWithWarnings={cellsWithWarningsState}
+          setCellsWithWarnings={(cellsData) => {
+            // eslint-disable-next-line no-console
+            console.log('cellsWithWarnings', cellsData);
+            setCellsWithWarningsState(cellsData);
+          }}
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.stories.tsx:180]: Тут тоже

File: src/components/vega-table/VegaTable.stories.tsx [hunk 3]
```diff
@@ -252,12 +282,21 @@ storiesOf('ui/VegaTable', module)
             console.log('selectedRows', indexes);
             setSelectedRowsState(indexes);
           }}
+          cellsWithWarnings={cellsWithWarningsState}
+          setCellsWithWarnings={(cellsData) => {
+            // eslint-disable-next-line no-console
+            console.log('cellsWithWarnings', cellsData);
+            setCellsWithWarningsState(cellsData);
+          }}
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.stories.tsx:290]: И тут

File: src/components/vega-table/VegaTable.tsx [hunk 4]
```diff
@@ -31,6 +31,8 @@ export const VegaTable = <
     newColumnIndex,
     selectedRows,
     setSelectedRows,
+    cellsWithWarnings,
+    setCellsWithWarnings,
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.tsx:35]: Убрать

File: src/components/vega-table/VegaTable.tsx [hunk 5]
```diff
@@ -92,6 +94,8 @@ export const VegaTable = <
       newColumnIndex,
       selectedRows,
       setSelectedRows,
+      cellsWithWarnings,
+      setCellsWithWarnings,
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.tsx:98]: Убрать

File: src/components/vega-table/VegaTable.tsx [hunk 6]
```diff
@@ -111,24 +115,24 @@ export const VegaTable = <
...
+    handleSetColumns,
+    columnHasContextMenu,
     summaryRowHeight,
     headerRowHeight,
     isReadOnly,
     newColumnIndex,
     selectedRows,
     setSelectedRows,
+    cellsWithWarnings,
+    setCellsWithWarnings,
```
Comment (SergeyKatugin) [inline src/components/vega-table/VegaTable.tsx:135]: Убрать

File: src/components/vega-table/context/VegaTableContext.tsx [hunk 7]
```diff
@@ -41,6 +42,8 @@ export type VegaTableAPI<
   newColumnIndex?: number | undefined;
   selectedRows?: number[];
   setSelectedRows?: (idx: number[]) => void;
+  cellsWithWarnings?: CellWithWarnings[];
+  setCellsWithWarnings?: (cellData: CellWithWarnings[]) => void;
```
Comment (SergeyKatugin) [inline src/components/vega-table/context/VegaTableContext.tsx:46]: Убрать

File: src/components/vega-table/context/VegaTableContext.tsx [hunk 8]
```diff
@@ -125,6 +129,8 @@ export const VegaTableContext = createContext<VegaTableAPI>({
   setFiltering: () => {},
   selectedRows: [],
   setSelectedRows: () => {},
+  cellsWithWarnings: [],
+  setCellsWithWarnings: () => {},
```
Comment (SergeyKatugin) [inline src/components/vega-table/context/VegaTableContext.tsx:133]: И тут

File: src/components/vega-table/types.ts [hunk 9]
```diff
@@ -141,3 +141,8 @@ export type EditorProps =
 export type EditorResult = {
   editor?: ComponentType<EditorProps>;
 };
+
+export type CellWithWarnings = {
```
Comment (SergeyKatugin) [inline src/components/vega-table/types.ts:145]: Не нужно

File: src/components/vega-table/VegaTable.tsx (hunk 6 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.tsx:135]: Готово

File: src/components/vega-table/context/VegaTableContext.tsx (hunk 7 above)
Comment (Alexandr1111) [reply src/components/vega-table/context/VegaTableContext.tsx:46]: Готово

File: src/components/vega-table/context/VegaTableContext.tsx (hunk 8 above)
Comment (Alexandr1111) [reply src/components/vega-table/context/VegaTableContext.tsx:133]: Готово

File: src/components/vega-table/types.ts (hunk 9 above)
Comment (Alexandr1111) [reply src/components/vega-table/types.ts:145]: Убрано

File: src/components/vega-table/VegaTable.stories.tsx (hunk 1 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx:68]: Готово

File: src/components/vega-table/VegaTable.stories.tsx (hunk 2 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx:180]: Убрано

File: src/components/vega-table/VegaTable.stories.tsx (hunk 3 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.stories.tsx:290]: Убрано

File: src/components/vega-table/VegaTable.tsx (hunk 4 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.tsx:35]: Убрано

File: src/components/vega-table/VegaTable.tsx (hunk 5 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.tsx:98]: Убрано

File: src/components/vega-table/VegaTable.tsx (hunk 6 above)
Comment (Alexandr1111) [reply src/components/vega-table/VegaTable.tsx:135]: Убрано

File: src/components/vega-table/context/VegaTableContext.tsx (hunk 7 above)
Comment (Alexandr1111) [reply src/components/vega-table/context/VegaTableContext.tsx:46]: Убрано

File: src/components/vega-table/context/VegaTableContext.tsx (hunk 8 above)
Comment (Alexandr1111) [reply src/components/vega-table/context/VegaTableContext.tsx:133]: Убрано

File: src/components/vega-table/types.ts (hunk 9 above)
Comment (Alexandr1111) [reply src/components/vega-table/types.ts:145]: Убрано

**Review by SergeyKatugin — APPROVED** [review]

**Pull request merged**